spec/Functions.tex

```diff
@@ -767,6 +764,43 @@ \subsection{The Ref Return Intent}
 \end{chapeloutput}
 \end{chapelexample}
 
+
+\subsection{The Const Ref Return Intent}
+\label{Const_Ref_Return_Intent}
+\index{functions!const ref keyword and@\chpl{const ref} keyword and}
+\index{const ref (return intent)@\chpl{const ref} (return intent)}
+\index{intents!const ref return@\chpl{const ref} return}
+
+The \chpl{const ref} return intent is also available. It is a restricted
+form of the \chpl{ref} return intent. In particular, it uses a similar
+mechanism but the function call cannot be assigned to. The \chpl{const ref}
+return intent should be viewed as a performance optimization. It
+indicates that the returned value does not need to be copied because the
+returned value will not be destroyed when the function returns.
+
+As with the \chpl{ref} return intent, it is a program error to return a
+variable with the \chpl{const ref} return intent if that variable is used
+after it is destroyed.
+
+\subsection{Choosing a Ref Return Intent Function Based on Calling Context}
+\label{Ref_Pair}
+\index{functions!ref-pair}
+
+During function resolution, if a procedure with a \chpl{ref} return
+intent and a procedure without a \chpl{ref} return intent are both
+possible candidates for a call, and either both use promotion or neither
+do, then this pair of procedures is called a ref-pair. Note that the
+procedure without \chpl{ref} return intent in a ref-pair could have no
+specified return intent, it could have the \chpl{const} return intent,
+or it could have the \chpl{const ref} return intent.
+
+When the call to a ref-pair is on the left-hand side of an
+assignment statement or appears in a call and corresponds to a formal argument
+with \chpl{out}, \chpl{inout}, or \chpl{ref} intent, the \chpl{ref} return
+intent procedure is used. Otherwise, the other candidate is used.  Thus,
+a ref-pair can be useful for adjusting the behavior depending on the
+calling context.
+
```

Comment:
I don't understand the motivation for introducing the term "ref-pair" and I don't think it's correct to talk
about calling a "ref pair".  The job of function resolution is to determine *the* function to call.  I think
this text is trying to discuss the manner in which the rules for determining the "most specific function"
needs to be extended to handle the case when the candidate functions include a function that returns with
ref intent and a different function that return with const ref intent.

I am not sure how to proceed at this late stage of the game but IMO most of the this text belongs in 13.12.3
(which is a beast) and then there might be a reference from here to there.